src/cli/mod.rs

```diff
@@ -48,7 +50,38 @@ pub struct Options {
 
 impl Options {
     pub fn execute(self) -> anyhow::Result<ExitCode> {
-        self.logger.init();
+        let env_filter = EnvFilter::builder()
+            .with_default_directive(self.logger.verbosity_level().into())
+            .parse_lossy(self.logger.log_filter.as_ref().unwrap_or(&"".to_owned()));
+        let fmt_layer = fmt::layer()
+            .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
+            .with_filter(env_filter);
+
+        let (flame_layer, _guard) = if let Some(ref profile) = self.logger.profile {
+            // Open `env::tempdir()` to reserve fd 3 on Unix, which `exec-wasmtime` will expect to read config from
+            // It will be dropped at the end of this block, freeing it for the following socketpair call
+            #[cfg(unix)]
+            let _reserved = matches!(
+                self.cmd,
+                Subcommands::Run(_)
+                    | Subcommands::Deploy(_)
+                    | Subcommands::Unstable(unstable::Subcommands::Exec(_))
+            )
+            .then_some(std::fs::File::open(std::env::temp_dir()))
```

Comment:
I still feel like we can do better, is there some path that is guaranteed to exist cross-platform and is fast to open?
What if we try and open `current_exe` or `current_dir`? Those are probably more likely to be there than `temp_dir`, although potentially still could be deleted once the executable is started?

@haraldh any ideas?

`/proc/self/exe` comes to mind, but I am not sure if that's really generic across all Unix'es

Maybe something to do with memory?

Was `/dev/null` used in the initial iteration the best solution?

All we really need is to get an fd 3 assigned by the kernel, we don't really care about anything on FS here.

Should we just open the `profile` path twice?

Reply:
/dev/null is fine